crates/router/src/core/payments.rs

```diff
@@ -1177,20 +1184,24 @@ pub async fn list_payments(
     merchant: domain::MerchantAccount,
     constraints: api::PaymentListConstraints,
 ) -> RouterResponse<api::PaymentListResponse> {
-    use futures::stream::StreamExt;
-
     use crate::types::transformers::ForeignFrom;
 
     helpers::validate_payment_list_request(&constraints)?;
     let merchant_id = &merchant.merchant_id;
-    let payment_intents =
+    let mut payment_intents =
         helpers::filter_by_constraints(db, &constraints, merchant_id, merchant.storage_scheme)
             .await
             .to_not_found_response(errors::ApiErrorResponse::PaymentNotFound)?;
-
-    let pi = futures::stream::iter(payment_intents)
-        .filter_map(|pi| async {
-            let pa = db
+    payment_intents.sort();
+    payment_intents.reverse();
```

Comment:
if possible we'd prefer to sort this via order clause in the diesel query...

That way the pagination would also be more user friendly